Frameworks/UIKit/UIImage.mm

```diff
@@ -44,487 +44,416 @@ @interface UIImage ()
-    IWICBitmapDecoder* pDecoder = NULL;
-    IStream* spStream = NULL;
-    IWICBitmapFrameDecode* pFrame = NULL;
-    HRESULT hr = S_OK;
-
-    MULTI_QI mq = { 0 };
-
-    mq.pIID = &IID_IWICImagingFactory;
-    hr = CoCreateInstanceFromApp(CLSID_WICImagingFactory, NULL, CLSCTX_INPROC_SERVER, NULL, 1, &mq);
-
-    if (SUCCEEDED(hr)) {
-        pFactory = (IWICImagingFactory*)mq.pItf;
-        hr = pFactory->CreateDecoder(decoderCls, NULL, &pDecoder);
-    }
-
-    if (SUCCEEDED(hr)) {
-        hr = ::CreateStreamOnHGlobal(NULL, FALSE, &spStream);
-        if (SUCCEEDED(hr)) {
-            ULONG written = 0;
-            hr = spStream->Write(bytes, length, &written);
-            if (SUCCEEDED(hr) && written == length) {
-                hr = pDecoder->Initialize(spStream, WICDecodeMetadataCacheOnLoad);
-
-                if (SUCCEEDED(hr)) {
-                    hr = pDecoder->GetFrame(0, &pFrame);
-                    if (SUCCEEDED(hr)) {
-                        if (!loadImageFromWICFrame(dest, pFactory, pFrame)) {
-                            TraceError(TAG, L"loadImageFromWICFrame failed");
-                            hr = E_FAIL;
-                        }
-                    } else {
-                        TraceError(TAG, L"IWICBitmapDecoder::GetFrame failed hr=%x", hr);
-                    }
-                } else {
-                    TraceError(TAG, L"IWICBitmapDecoder::Initialize failed hr=%x", hr);
-                }
-            } else {
-                TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr, length, written);
-            }
-        } else {
-            TraceError(TAG, L"CreateStreamOnHGlobal failed hr=%x", hr);
-        }
-    }
-
-    if (pFactory) {
-        pFactory->Release();
-    }
-
-    if (pDecoder) {
-        pDecoder->Release();
-    }
-
-    if (spStream) {
-        spStream->Release();
-    }
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)image {
+  RETURN_NULL_IF(!image);
+  if (self = [self init]) {
+    m_pImage = image;
+    CGImageRetain(image);
+  }
+  return self;
+}
 
-    if (pFrame) {
-        pFrame->Release();
-    }
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)image
+                          scale:(float)scale
+                    orientation:(UIImageOrientation)orientation {
+  if (self = [self initWithCGImage:image]) {
+    _scale = scale;
+    _orientation = orientation;
+  }
 
-    return SUCCEEDED(hr);
+  return self;
 }
 
-static bool loadGIF(UIImage* dest, void* bytes, int length) {
-    return loadImageWithWICDecoder(dest, GUID_ContainerFormatGif, bytes, length);
-}
+- (instancetype)_initWithCopyOfImage:(UIImage *)imageToCopy
+                   WithRenderingMode:(UIImageRenderingMode)renderingMode {
 
-static bool loadBMP(UIImage* dest, void* bytes, size_t length) {
-    return loadImageWithWICDecoder(dest, GUID_ContainerFormatBmp, bytes, length);
-}
+  RETURN_NULL_IF(!imageToCopy);
```

Comment:
this comment seemed misplaced.

---
In reply to: [84145437](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84145437) [](ancestors = 84145437)